code/game/turfs/turf.dm

```diff
@@ -359,23 +362,50 @@
 	return FALSE
 
 /turf/open/snow/is_weedable()
-	return !slayer
-
-/turf/open/mars/is_weedable()
-	return TRUE
-
+	return !slayer && ..()
 
 /turf/open/floor/plating/plating_catwalk/is_weedable() //covered catwalks are unweedable
+	. = ..()
 	if(covered)
 		return FALSE
-	else
-		return TRUE
 
 
 /turf/closed/wall/is_weedable()
-	return TRUE //so we can spawn weeds on the walls
-
+	return !is_type_in_typecache(get_area(src), GLOB.unweedable_areas) //so we can spawn weeds on the walls
+
+/turf/proc/check_alien_construction(mob/living/L)
+	var/has_obstacle
+	for(var/obj/O in contents)
+		if(istype(O, /obj/item/clothing/mask/facehugger))
```

Comment:
I wonder here what's more efficient, if redefined procs or these multiple checks.
@MrStonedOne seemed to suggest proc overhead was worse, while @ninjanomnom suggested multiple checks to be less desirable.